Getting to release

McWhity opened the pull request:
Please check if their are any problems with the merge

TonioF commented on sar_pre_processing/file_list_sar_pre_processing.py:
```diff
@@ -84,34 +63,29 @@ def _select_year(self, filelist, year):
         filelist_new = []
         for file in filelist:
             filepath, filename, fileshortname, extension = self._decomposition_filename(
                 file)
```
merge into previous line

McWhity replied:
Done

TonioF commented on sar_pre_processing/netcdf_stack.py:
```diff
@@ -84,108 +82,48 @@ def _create_empty_netcdf_file(self):
+                sat = 0
+            elif data.satellite == 'S1B':
+                sat = 1
+
```
It would be good to throw an exception in the else case

McWhity replied:
Else case added.
value was set to -9999 (missing value)

TonioF commented on sar_pre_processing/netcdf_stack.py:
```diff
@@ -84,108 +82,48 @@ def _create_empty_netcdf_file(self):
+                orbitdir = 0
+            elif data.orbitdirection =='DESCENDING':
+                orbitdir = 1
 
```
It would be good to throw an exception in the else case

McWhity replied:
Else case added.
value was set to -9999 (missing value)

TonioF commented on sar_pre_processing/sample_config_file.yml:
```diff
@@ -1,63 +1,43 @@
 
```
Can we please remove this file? It looks bad to have something with absolute paths to your personal computer on github. Also, this file is not needed anywhere.

McWhity replied:
agreed

TonioF commented on sar_pre_processing/sar_pre_processor.py:
```diff
@@ -158,426 +162,381 @@ def pre_process_step1(self, **kwargs):
 
-        !!! all files will get metadata of the master image !!! Problem?
-
+        !!! all files will get metadata of the master image !!! That is how SNAP does it! Metadata will be corrected within netcdf output files at the end of the preprocessing chain (def add_netcdf_information)
```
break line

McWhity replied:
done

TonioF commented on sar_pre_processing/sar_pre_processor.py:
```diff
@@ -158,426 +162,381 @@ def pre_process_step1(self, **kwargs):
-                                else:
-                                    pass
+                                orbit_dir = iiii.text
+                                if orbit_dir == 'ASCENDING':
```
These lines make no sense. Also, I think it is better to handle the else case - as it was before.

TonioF replied:
I still think so - or can you explain why you are assigning to a variable a value that this variable already holds?

McWhity replied:
you are right. lets remove it

TonioF commented on sar_pre_processing/sar_pre_processor.py:
```diff
@@ -158,426 +162,381 @@ def pre_process_step1(self, **kwargs):
-if __name__ == "__main__":
-    processing = SARPreProcessor(config='sample_config_file.yml')
-    processing._create_processing_filelist()
+# if __name__ == "__main__":
```
Remove this

TonioF replied:
I meant: Can you remove the commented main? It's always confusing to have old pieces of code lying around. If you want to keep it, at least add one more line why it is still there.

McWhity replied:
Thanks for clarification. It is removed

TonioF commented on setup.py:
```diff
@@ -1,11 +1,17 @@
 #!/usr/bin/env python
 
 from setuptools import setup
+import os
```
No need to import os

McWhity replied:
removed

TonioF commented on setup.py:
```diff
@@ -1,11 +1,17 @@
 
+# removed dependencies to prevent multiple installations
 requirements = [
-
```
remove requirements and put everything into environment.yml instead.

McWhity replied:
done

TonioF commented on README.md:
```diff
@@ -1,2 +1,68 @@
+<!-- [![Build Status](https://travis-ci.org/McWhity/sar-pre-processing.svg?branch=master)](https://travis-ci.org/McWhity/sar-pre-processing) -->
+[![Documentation Status](https://readthedocs.org/projects/multiply-sar-pre-processing/badge/?version=getting_to_release)](https://multiply-sar-pre-processing.readthedocs.io/en/getting_to_release/?badge=getting_to_release)
+
+This repository contains the functionality for Sentinel-1 SAR-Pre.Processing of the MULTIPLY main platform.
```
SAR-Pre-Processing

TonioF replied:
What I meant: There is a typo. SAR-Pre.Processing -> SAR-Pre-Processing

TonioF commented on README.md:
```diff
@@ -1,2 +1,68 @@
+* `test/` - The test package.
+* `setup.py` - main build script, to be run with Python 3.6
+* `LICENSE.rst` - License of software in repository.
+to be updated!!!!
```
No, it has been updated.

McWhity replied:
Dont't know what you mean with your comment

TonioF replied:
Me neither. I can't remember.

TonioF commented on README.md:
```diff
@@ -1,2 +1,68 @@
+
+## Usage
+
+For usage checkout the [juypter notebook](https://nbviewer.jupyter.org/github/McWhity/sar-pre-processing/tree/getting_to_release/docs/notebooks/)
```
Did you mean to link to your private repo?

McWhity replied:
No. I did not update the link. Have done it now

TonioF commented on README.md:
```diff
@@ -1,2 +1,68 @@
+## Contribution and Development
+
+Once, the package is set up, you are very welcome to contribute to MULTIPLY SAR-Pre-Processing.
+
```
=> You are very welcome to contribute to MULTIPLY SAR-Pre-Processing. To do so, please first make a fork into your own repository and then create a Pull Request.

McWhity replied:
replaced

TonioF commented on README.md:
```diff
@@ -1,2 +1,68 @@
+
+## License
+
+This project is licensed under the GPLv3 License - see the [LICENSE.md](https://github.com/multiply-org/sar-pre-processing/blob/master/LICENSE.rst) file for details.
```
LICENSE.md -> License.rst

McWhity replied:
Thanks for the hint. Done

TonioF commented on docs/04_Theory.rst:
```diff
@@ -1,30 +1,27 @@
-The two Sentinel-1 satellites 1A and 1B are one of the first satellites which are providing microwave data in high temporal and spatial resolution. Within the MULTIPLY project we developed a preprocessing chain to process time-series of Sentinel-1 data for quantitative analysis of vegetation and soil parameters over agricultural fields. Therefore, rigorous geometric and radiometric corrections as ...
-
-The whole preprocessing chain for Sentinel-1 Level-1 Single Look Complex (SLC) data is accomplished by ESA's SNAP S1TBX software (current version 5.0.4). The SNAP toolbox can be downloaded from `<http://step.esa.int/main/download/>`_. However, to automatically apply different preprocessing steps on Sentinel-1 data a python script, which uses the Graph Processing Tool (GPT) of the S1TBX, is provid ...
+The different preprocessing steps are shown in :numref:`work_flow_step_1` and :numref:`work_flow_step_2`. Additionally, every processing step is explained in more detail in the following subsections. As it can be seen in :numref:`work_flow_step_1` and :numref:`work_flow_step_2` the preprocessing work-flow is split in two main parts. The preprocessing methods in :numref:`work_flow_step_1` can be a ...
```
"need several images" -> "needs several images"

McWhity replied:
Done

TonioF commented on docs/04_Theory.rst:
```diff
@@ -233,14 +230,36 @@ A characteristic of images acquired by a SAR system is the visibility of random
 Practical implementation
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-For the speckle reduction the "Multi-temporal Speckle Filter" operator within SNAP's S1TBX software is used. Currently 15 temporally consecutive images are used within the "Multi-temporal Speckle Filter" whereby the target image is temporally situated in the middle. The applied filter is a Lee filter with spatial averaging over 3x3 pixel. The spatial averaging over pixel has a significant influen ...
+For the speckle reduction the "Multi-temporal Speckle Filter" operator within SNAP's S1TBX software is used. Currently 7 temporally consecutive images are used within the "Multi-temporal Speckle Filter" whereby the target image is temporally situated in the middle. The applied filter is a Lee filter with a spatial window size of 5x5 pixels, a sigma of 0.9, and a target window size of 3x3 pixels.  ...
```
Currently 7 temporally consecutive ... -> As default, 7 temporally consecutive

McWhity replied:
Agreed. Done

TonioF commented on recipe/meta.yaml:
```diff
@@ -4,9 +4,10 @@ about:
 
 package:
   name: multiply-sar-preprocessing
-  version: "0.1"
```
We should use version 0.1 here, too. Otherwise, set it to 0.4 in version.py and changes.md, but we need the same version everywhere.

McWhity replied:
Agreed. Changed to 0.1

TonioF reviewed (changes requested):
- In the readme, we need to state that SNAP needs to be installed as a prerequisite.
- Please remove comments_tonio.doc completely
- Rename flow_chard.png to flow_chart.png
- Rename flow_chard_01.png to flow_chart_01.png
- In general, we should set up the environment in the environment.yml. Put all required dependencies there instead of installing them in setup.py, then create your dedicated environment.
- In the Jupyter notebook, you can then remove the parts where you install gdal and sentinelsat
- Also, in the notebook you state that gpt is usually found around '/home/snap/bin/gpt'. Better: gpt is located in the bin folder of your SNAP installation

McWhity commented:
Regrading:
>     * In the readme, we need to state that SNAP needs to be installed as a prerequisite.
information was added

>     * Please remove comments_tonio.doc completely
Done
 
>     * Rename flow_chard.png to flow_chart.png
Cannot find the file !

>     * Rename flow_chard_01.png to flow_chart_01.png
Cannot find the file !

>     * In general, we should set up the environment in the environment.yml. Put all required dependencies there instead of installing them in setup.py, then create your dedicated environment.
I put a lot of time to get the installation working. I'm not an expert in this. If there is a better way you are very welcome to help me set it up. Thanks
 
>     * In the Jupyter notebook, you can then remove the parts where you install gdal and sentinelsat
Sentinelsat is not part of the sar-pre-processing. But to run the sar-pre-processing you need example data. This is why the Sentinelsat package is installed within the notebook and not within the package itself.
I couldn't get the installation of gdal work within the package. Therefore I introduced a little workaround. If you know a better way please let me now.
 
>     * Also, in the notebook you state that gpt is usually found around '/home/snap/bin/gpt'. Better: gpt is located in the bin folder of your SNAP installation
Changed

TonioF commented:
I have added an environment.yml. You can use miniconda (recommended) or anaconda to easily create an environment from this. Do you want me to show you how to do this?

TonioF commented on docs/index.rst:
```diff
@@ -1,18 +1,50 @@
+    :target: https://travis-ci.org/multiply-org/sar-pre-processing
 
+.. |docstatus| image:: https://readthedocs.org/projects/multiply-sar-pre-processing/badge/?version=latest
+    :target: https://multiply-sar-pre-processing.readthedocs.io/en/getting_to_release/
```
```suggestion
    :target: https://multiply-sar-pre-processing.readthedocs.io/en/getting_to_release/?badge=getting_to_release
```

McWhity replied:
Done

TonioF commented on README.md:
```diff
@@ -1,2 +1,79 @@
+
+## Authors
+
+{{AUTHORS.rst}}
```
Just list our names here.

McWhity replied:
relative link to file included

TonioF commented on docs/notebooks/running_test_application.ipynb:
```diff
@@ -0,0 +1,354 @@
+    "## 1. Requirements\n",
+    "\n",
+    "- Installation of sar-pre-processing package\n",
+    "    - Problem: gdal package might not be installed correctly\n",
```
I did not have this problem. SNAP was able to use its built-in gdal package. I recommend that you either remove this sentence or state what should be done if this problem occurs.

McWhity replied:
Old Problem. Statement removed

TonioF commented on docs/notebooks/running_test_application.ipynb:
````diff
@@ -0,0 +1,354 @@
+    "- Installation of ESA's SNAP Toolbox (https://step.esa.int/main/download/snap-download/)\n",
+    "    - Update SNAP Toolbox after installing for fixing bugs within the software\n",
+    "    - SNAP Toolbox need libgfortran for specific operations but currently libgfortran is not installed during the installation process of SNAP therefore you might use\n",
+    "```sudo apt-get install gfortran```\n",
````
When on linux

McWhity replied:
included

TonioF commented on docs/notebooks/running_test_application.ipynb:
```diff
@@ -0,0 +1,354 @@
+    "\n",
+    "data['input_folder'] = input_folder\n",
+    "data['output_folder'] = output_folder\n",
+    "data['gpt'] = gpt_location\n",
```
I'd love if you'd state how to change more parameters, namely year and bounding box.

McWhity replied:
included

TonioF commented on docs/notebooks/running_test_application.ipynb:
```diff
@@ -0,0 +1,354 @@
+    "    print(\"matplotlib installed - requirement satisfied.\")\n",
+    "except ModuleNotFoundError:\n",
+    "    print('matplotlib module not installed!')\n",
+    "    yes = input(\"Do you want do install now? (type y)\")\n",
```
This did not work in my installation. Also, when using conda we should avoid using pip whenever possible. Either add matplotlib as requirement in environments.yml or state at the beginning of the notebook that users should execute 'conda install -c conda-forge matplotlib' first.

McWhity replied:
pip statement removed, conda statement included

TonioF commented on docs/notebooks/running_test_application.ipynb:
```diff
@@ -0,0 +1,354 @@
+    "from netCDF4 import Dataset\n",
+    "import numpy as np\n",
+    "\n",
+    "my_example_nc_file = os.path.join(output_folder + 'data.nc')\n",
```
In my notebook, I had to change this. Maybe because I did not state my output directory correctly? It ended with '/test', without a final slash. 

```suggestion
    "    "my_example_nc_file = os.path.join(output_folder + '/test.nc')\n",
",
```
